internal/controllers/rest_controller.go

```diff
@@ -0,0 +1,122 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/coreos/go-oidc"
+	"github.com/gin-gonic/gin"
+	"golang.org/x/oauth2"
+
+	"github.com/ydataai/authentication-service/internal/clients"
+	"github.com/ydataai/authentication-service/internal/services"
+	"github.com/ydataai/go-core/pkg/common/logging"
+	"github.com/ydataai/go-core/pkg/common/server"
+)
+
+// RESTController defines rest controller
+type RESTController struct {
+	restService   *services.RESTService
+	configuration RESTControllerConfiguration
+	oidcClient    *clients.OIDCClient
+	logger        logging.Logger
+}
+
+// NewRESTController initializes rest controller
+func NewRESTController(restService *services.RESTService,
+	configuration RESTControllerConfiguration,
+	oidcClient *clients.OIDCClient,
+	logger logging.Logger) RESTController {
+	return RESTController{
+		restService:   restService,
+		configuration: configuration,
+		oidcClient:    oidcClient,
+		logger:        logger,
+	}
+}
+
+// Boot ...
+func (rc RESTController) Boot(s *server.Server) {
+	s.Router.GET("/healthz", rc.healthCheck())
+
+	s.Router.GET(rc.configuration.AuthServiceURL, gin.WrapF(rc.RedirectAuthEndpoint))
+	s.Router.GET(rc.configuration.OIDCCallbackURL, gin.WrapF(rc.Callback))
+	// s.Router.POST(rc.configuration.LogoutURL, gin.WrapF(rc.Logout))
+}
+
+func (rc RESTController) healthCheck() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		ctx.Status(http.StatusNoContent)
+	}
+}
+
+// RedirectAuthEndpoint is the handler responsible for redirecting to the auth endpoint.
+func (rc RESTController) RedirectAuthEndpoint(w http.ResponseWriter, r *http.Request) {
```

Comment:
i would call this method something more clear, redirect is the action, not everything that it does 🤔